datahub-web-react/src/app/shared/tags/AddTagModal.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import { FetchResult } from '@apollo/client';
+import { Button, Modal, Select, Typography } from 'antd';
+import styled from 'styled-components';
+
+import { UpdateDatasetMutation } from '../../../graphql/dataset.generated';
+import { useGetAutoCompleteResultsLazyQuery } from '../../../graphql/search.generated';
+import { GlobalTags, GlobalTagsUpdate, EntityType, TagAssociationUpdate } from '../../../types.generated';
+import { convertTagsForUpdate } from './utils/convertTagsForUpdate';
+import CreateTagModal from './CreateTagModal';
+
+type AddTagModalProps = {
+    globalTags?: GlobalTags | null;
+    updateTags?: (
+        update: GlobalTagsUpdate,
+    ) => Promise<FetchResult<UpdateDatasetMutation, Record<string, any>, Record<string, any>>>;
```

Comment:
Does this modal need to know the return type here? Can it be void to simplify?

Reply:
The modal needs this so it can show/hide loading information